blueprints/addon/files/tsconfig.declarations.json

```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "declarationDir": "declarations",
+    "emitDeclarationOnly": true,
+    "isolatedModules": false,
+    "noEmit": false
+  },
+  "include": ["addon/**/*", "types/**/*", "addon-test-support/**/*"]
```

Comment:
Should we be including `types` here? Since that's the equivalent of v2's `unpublished-development-types` directory, it seems to me like we'd want to ensure the addon's declarations can be successfully generated without relying on declarations in there, but I could be missing something?

(Also, as an aside: I think we can drop the `**/*` here and just use the directory names. IIRC requiring the globs was something we had to do to account for a weird corner case in `tsc`'s FS watcher back in `ember-cli-typescript@1.x` when we used `tsc` as part of the build to do transpilation.)